servicetalk-http-api/src/main/java/io/servicetalk/http/api/StreamingHttpClientFilterFactory.java

```diff
@@ -37,7 +39,13 @@
      *
      * @param <U> the type of address before resolution (unresolved address).
      * @return a {@link MultiAddressHttpClientFilterFactory} function
+     * @deprecated Use
+     *   {@link MultiAddressHttpClientBuilder#initializer(MultiAddressHttpClientBuilder.SingleAddressInitializer)} and
+     *   {@link SingleAddressHttpClientBuilder#appendClientFilter(StreamingHttpClientFilterFactory)}
+     *   on the last argument of
+     *   {@link SingleAddressInitializer#initialize(String, Object, SingleAddressHttpClientBuilder)}.
      */
+    @Deprecated
```

Comment:
Following @Scottmitch comment from https://github.com/apple/servicetalk/pull/1721#discussion_r685187056 I also added deprecation for this method. Please re-review @tkountis @Scottmitch.